engine/src/main/java/com/sandboni/core/engine/finder/bcel/visitors/CallerClassVisitor.java

```diff
@@ -36,14 +38,39 @@ public void visitMethod(Method method) {
 
         // in order to detect locations (modules) each method has to have at least one entry in the graph as caller
         if (linksAdded == 0) {
-            Vertex v ;
-            if (Objects.nonNull(context.getCurrentLocation()) && context.getTestLocations().contains(context.getCurrentLocation())){
+            Vertex v;
+            if (isTestLocation()) {
                 v = new TestVertex.Builder(this.javaClass.getClassName(), formatMethod(method), context.getCurrentLocation()).build();
-            }else{
+            } else {
                 v = new Vertex.Builder(this.javaClass.getClassName(), formatMethod(method), context.getCurrentLocation())
                         .build();
             }
-            context.addLink(LinkFactory.createInstance(context.getApplicationId(), v , DEAD_END_VERTEX, LinkType.METHOD_CALL));
+            context.addLink(LinkFactory.createInstance(context.getApplicationId(), v, DEAD_END_VERTEX, LinkType.METHOD_CALL));
         }
     }
+
+    private boolean isTestLocation() {
+        return Objects.nonNull(context.getCurrentLocation()) && context.getTestLocations().contains(context.getCurrentLocation());
+    }
+
+    @Override
+    public void visitConstantPool(final ConstantPool obj) {
+        if (ClassUtils.foundReflectionInMethodCalls(obj)) {
+            linkToReflectionCallVertrx();
+        }
+    }
+
+    private void linkToReflectionCallVertrx() {
+        Vertex v;
+        LinkType linkType;
+        if(isTestLocation()) {
+            v = new TestVertex.Builder(this.javaClass.getClassName(), "reflection ref (test)", context.getCurrentLocation()).build();
```

Comment:
Shouldn't use the method name instead of fixed strings?

Reply:
I don't have the method name here. this is just checking if there is any invocation in the visited class that makes a reflection call.